src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -3201,6 +3202,10 @@ private function registerRateLimiterConfiguration(array $config, ContainerBuilde
             $limiter->replaceArgument(0, $limiterConfig);
 
             $container->registerAliasForArgument($limiterId, RateLimiterFactory::class, $name.'.limiter');
+
+            if (interface_exists(RateLimiterFactoryInterface::class)) {
```

Comment:
Is the `interface_exists` here required?

Reply:
Yes if FWB is installed without the RateLimiter component

Reply:
I think the intention is to check if the currently installed version of rate-limiter includes this new interface.

I'm questioning it because: if the interface isn't available and we don't have this check. I think it'll just be a string that won't break anything.

It would add to the `debug:autowiring` command so perhaps that reason enough to keep this logic.

Reply:
Even if it doesn't break the container itself, the `debug:autowiring` argument seems good enough to me to check the interface. It would be very misleading to see the interface as available for autowiring, but breaking the app when trying to use it

Reply:
Agreed!